File: internal/parser/java/testdata/fixture/src/main/java/com/example/demo/controller/UserController.java
package com.example.demo.controller;

import com.example.demo.model.User;
import com.example.demo.service.UserService;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/users")
public class UserController {
    @Autowired
    private UserService userService;

    @GetMapping("/{id}")
    public User getUser(Long id) {
        return this.userService.getUser(id);
    }

    @DeleteMapping(
        "/{id}"
    )
    public void deleteUser(Long id) {
    }
}

File: internal/parser/typescript/testdata/fixture/src/user.controller.ts
import { User } from './user.entity';
import { UserService } from './user.service';

@Controller('users')
export class UserController {
    @Inject()
    private userService: UserService;

    @Get(':id')
    getUser(id: number): User {
        return this.userService.getUser(id);
    }

    @Post()
    createUser(): User {
        return this.userService.getUser(0);
    }

    @Delete(
        ':id'
    )
    deleteUser(id: number): void {
    }
}
